lib/review-apps.unit.test.js

```diff
@@ -43,23 +114,19 @@ describe('review-apps', () => {
 			reviewAppsNockScope.post('/review-apps')
 				.reply(200, {});
 
-			return createReviewApp({
-				pipelineId,
-				branch,
-				repoName,
-				commit,
-				githubToken
-			})
-				.then(async res => {
-					const { ok } = res;
-					expect(ok).toBeTruthy();
-
-					const reviewApp = await res.json();
-					expect(reviewApp).toBeTruthy();
-				});
+			const response = await createReviewApp({ pipelineId, branch, repoName, commit, githubToken });
+
+			const { ok } = response;
+
+			const reviewApp = await response.json();
+
+			expect(ok).toBeTruthy();
+			expect(reviewApp).toBeTruthy();
+
 		});
 
 		it('returns an error', async () => {
```

Comment:
This test is the same as `returns created review app`

Reply:
Good spot! I was only editing this test to use `await` (i.e. not `.then()`) and did not notice that it is not actually testing for an error to be returned (also unclear if this means an error should be returned or thrown).

I will amend it to test what is in its description.

Reply:
Having looked at this test and the corresponding function, while ostensibly it only tests that Nock is working, I think it is important to test that this function does not throw an error in the instance of the API returning a 400.

There is a function in `review-apps.js` - `performFetch` - that all other data fetches use, and which will throw an error if the `ok` value in the response is false.

It is especially important for this specific function that it does not use `performFetch` so that an error is not thrown (as there is logic that needs to be performed in the event of a non-OK response), and so I think that justifies keeping it (though I have removed the expectation where it expects `error` from this line (which is just an empty object `{}`) to be truthy: `const error = await response.json();`).